packages/frontend/src/components/SiteWideBanner/index.tsx

```diff
@@ -1,28 +1,51 @@
-import { useCallback, useEffect, useState } from 'react'
+import { useCallback, useContext, useEffect, useState } from 'react'
 import CloseIcon from '@mui/icons-material/Close'
 import { Box, IconButton } from '@mui/material'
+import { LaunchDarklyContext } from 'contexts/LaunchDarkly'
 import { getItemForSession, setItemForSession } from 'helpers/storage'
-import useFormatMessage from 'hooks/useFormatMessage'
 
-const BANNER_TEXT_ID = 'bannerText'
+const LAUNCH_DARKLY_BANNER_KEY = 'app_banner_display'
+const EMPTY_BANNER_MESSAGE = ''
 
 const SiteWideBanner = (): JSX.Element | null => {
-  const formatMessage = useFormatMessage()
-  // seems like this cant return empty string, so using _ as hacky empty string alternative
-  const message = formatMessage(BANNER_TEXT_ID)
+  const [bannerMessage, setBannerMessage] = useState('')
   const [showBanner, setShowBanner] = useState(false)
+
+  const launchDarkly = useContext(LaunchDarklyContext)
+
   useEffect(() => {
-    const bannerTextStored = getItemForSession('hide-banner')
-    setShowBanner(message !== bannerTextStored)
-  }, [])
+    const bannerMessageStored = getItemForSession('banner-text')
+    // either banner should never be displayed based on LD or banner is 'closed'
+    setShowBanner(
+      !!bannerMessageStored && bannerMessageStored !== EMPTY_BANNER_MESSAGE,
+    )
+  }, [showBanner])
 
+  // TODO: check why use callback when there are no dependencies?
   const closeBanner = useCallback(() => {
     setShowBanner(false)
-    setItemForSession('hide-banner', message)
+    setItemForSession('banner-text', EMPTY_BANNER_MESSAGE)
   }, [])
 
-  // this means the banner text is empty, so dont show the banner
-  if (message === BANNER_TEXT_ID) {
+  // check for feature flag (takes time to load) to display banner: by default it should be not visible
+  useEffect(() => {
+    if (
+      launchDarkly.flags &&
+      launchDarkly.flags[LAUNCH_DARKLY_BANNER_KEY] !== 'none' // this is the value set in LD
```

Comment:
why use a special 'none'? u can set an empty string on launch darkly

Reply:
i thought that an empty string was not meaningful enough, and i stored the message inside because I thought that the intended effect was to only bring back the banner when someone restarts the session?